src/odemis/acq/align/test/align_test.py

```diff
@@ -0,0 +1,203 @@
+
+
+    def test_slit_cancel(self):
+        """
+        Test cancelling does cancel (relatively quickly)
+        """
+        bl = self.light
+        ccd = mock.FakeCCD(0, self.img_light_off)
+        f = turnLightAndCheck(bl, ccd)
+        time.sleep(1)
+        f.cancel()
+        self.assertTrue(f.cancelled())
+        with self.assertRaises(CancelledError):
+            f.result(5)
+
+    def test_1pixel_image(self):
+        """
+        Test from dark image to bright image of 1 pixel
+        """
+        bl = self.light
+
+        #dark 1x1 image
+        img_dark = numpy.zeros((1,1), dtype= numpy.uint8)  # type: ndarray
+        da_dark = model.DataArray(img_dark)
+        ccd = mock.FakeCCD(0, da_dark)
+
+        f = turnLightAndCheck(bl, ccd)
+        time.sleep(3)
+        self.assertFalse(f.done())
+
+        # simulate light turning on
+        # bright 1x1 image
+        img_bright = numpy.ones((1,1), dtype=numpy.uint8)
+        br_bright = model.DataArray(img_bright)
+        ccd.fake_img = br_bright
+
+        time.sleep(3)
+        self.assertTrue(f.done())
+        f.result()
+
+    def test_light_already_on(self):
+        """
+        Test when chamber light is already on
+        """
+        bl = self.light
+
+        # bright image - light on
+        self.data = tiff.read_data(os.path.join(TEST_IMAGE_PATH, "brightlight-on-slit-spccd-simple.ome.tiff"))
+        self.bright_image = img.ensure2DImage(self.data[0])
+        ccd = mock.FakeCCD(0, self.bright_image)
+
+        f = turnLightAndCheck(bl, ccd)
+        time.sleep(3)
+
+        self.assertTrue(f.done())
+        f.result()
+
+
+    def test_light_on_and_on(self):
+        """
+        Test when chamber light is already on and we take one brighter image
+        """
+        bl = self.light
+
+        # bright image - light on
+        self.data = tiff.read_data(os.path.join(TEST_IMAGE_PATH, "brightlight-on-slit-spccd-simple.ome.tiff"))
+        self.bright_image = img.ensure2DImage(self.data[0])
+        ccd = mock.FakeCCD(0, self.bright_image)
+
+        f = turnLightAndCheck(bl, ccd)
+        time.sleep(3)
+
+        self.assertTrue(f.done())
+
+        # bright image - light on
+        self.data = tiff.read_data(os.path.join(TEST_IMAGE_PATH, "brightlight-on-slit-ccd.ome.tiff"))
+        self.bright_image2 = img.ensure2DImage(self.data[0])
+        ccd.fake_img = self.bright_image2
+
+        time.sleep(1)
+        self.assertTrue(f.done())
+        f.result()
+
+    def test_no_detection_of_change(self):
+        """
+        Test when we get a new dark image with a bright spot but no change is detected
+        """
+        bl = self.light
+
+        ccd = mock.FakeCCD(0, self.img_light_off)
+
+        f = turnLightAndCheck(bl, ccd)
+        time.sleep(1)
+        self.assertFalse(f.done())
+
+        # take a dark image with a bright spot
+        data = hdf5.read_data(os.path.join(TEST_IMAGE_PATH, "one_spot.h5"))
```

Comment:
In general, you should avoid passing images which are from different detectors, as they have different shape, and different background. It's not realistic, and the code is not supposed to handle these cases, so failing/passing has no real meaning. I'd suggest to just use the light_off image, and set on a couple of pixels that maximum (or near maximum values).